src/fs_sys.c

```
@@ -0,0 +1,342 @@
+        return ERR_FDS_EXCEEDED;
+    }
+    int imode = mode_str_to_int(mode);
+    if (imode == 0) {
+        return ERR_BAD_MODE;
+    }
+
+    //strip path name for media
+    if (path[0] != '/') {
+        return ERR_BAD_PATH;
+    }
+    if (path[1] != '0' && path[1] != '1' && path[1] != '2' && path[1] != '3') {
+        return ERR_BAD_PATH;
+    }
+    if (path[2] != '/') {
+        return ERR_BAD_PATH;
+    }
+
+    char media_path[strlen(path) - 1];   //Two fewer than whole path length, + 1 for null terminator
+    strcpy(media_path, path+2);
+
+    int ata_unit = path[1] - 48; //48 is ascii for 0
+
+    enum ata_kind ata_type = map_media_to_driver_id(ata_unit);
+
+    int next_fd;
+    //find current's next available fd by 'files' member
+    //add to process file->file table
+    bool opened = 0;
+    for (next_fd = 0; opened == 0 && next_fd < PROCESS_MAX_OPEN_FILES; next_fd++) {
+        if (current->files->open_files[next_fd] == 0) {
+            //make new sys level file
+            //mark it with the ata type
+            //the ata unit
+            //and a pointer to the union of below
+            //sys level file types
+            current->files->open_files[next_fd] = create_fs_agnostic_file(ata_type, ata_unit, media_path, imode);
+            if (current->files->open_files[next_fd]) {
+                opened = 1;
+                current->files->num_open += 1;
+                break;
+            } else {
+                return ERR_KERNEL_OPEN_FAIL;
+            }
+        }
+    }
+
+    // TODO: keep on OS's open file table
+
+    return next_fd;
+}
+
+uint32_t fs_sys_close(uint32_t fd) {
+    if (fd >= PROCESS_MAX_OPEN_FILES) {
+        return ERR_FD_OOR;
+    }
+
+    struct fs_agnostic_file *fp = current->files->open_files[fd];
+    if (fp) {
+        switch (fp->ata_type) {
+            case 1:  //ISO
+                iso_fclose((struct iso_file *)fp);
+                break;
+            default:
+                break;
```

Should we return an error indicating an invalid type here, just in case?